daemon/execdriver/native/create.go

```diff
@@ -244,6 +282,18 @@ func (d *driver) setupMounts(container *configs.Config, c *execdriver.Command) e
 			flags |= syscall.MS_SLAVE
 		}
 
+		// for user-namespaces, any bind-mounted content which is owned by root needs
+		// to be remapped to the container's root, if user namespacing is turned on
+		srcStat, err := system.Stat(m.Source)
+		if err != nil {
+			return err
+		}
+		if c.RemappedRoot.Uid != 0 && srcStat.Uid() == 0 {
+			if err := os.Chown(m.Source, c.RemappedRoot.Uid, c.RemappedRoot.Gid); err != nil {
```

Comment:
Are you changing the directory/file's permissions on the host? Is that desirable?